src/main/java/io/prometheus/cloudwatch/CloudWatchCollector.java

```diff
@@ -640,6 +640,7 @@ private void scrape(List<MetricFamilySamples> mfs) throws CloneNotSupportedExcep
         }
         
         // Add the "aws_resource_info" metric for existing tag mappings
+        List<MetricFamilySamples.Sample> samples = new ArrayList<MetricFamilySamples.Sample>();
```

Comment:
This needs to be at the top of the function, before the for loop. A name like infoSamples would also be clearer.